File: frontend/src/types/calendar.ts
export interface Calendar {
  id: string;
  name: string;
  color: string;
  created_at: string;
}

export interface CreateCalendarInput {
  name: string;
  url?: string;
  google_id?: string;
  color?: string;
}

export interface GoogleCalendarEntry {
  id: string;
  summary: string;
  backgroundColor?: string;
  primary?: boolean;
}

File: backend/src/models/calendar.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Calendar {
    pub id: Uuid,
    pub name: String,
    pub url: Option<String>,
    pub google_id: Option<String>,
    pub color: String,
    pub created_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct CalendarPublic {
    pub id: Uuid,
    pub name: String,
    pub color: String,
    pub created_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Deserialize, Serialize)]
pub struct CreateCalendarSchema {
    pub name: String,
    pub url: Option<String>,
    pub google_id: Option<String>,
    pub color: Option<String>,
}
